File: Python/samples/upload_example.py
"""上传文件示例"""
import os
from samples.example_base import ExampleBase
from ilinkai_weixin import WeixinApiClient, WeixinApiClientOptions
from ilinkai_weixin.auth.account_store import AccountStore
from ilinkai_weixin.media.media_upload_service import MediaUploadService
from ilinkai_weixin.weixin_api_client import DEFAULT_BASE_URL, DEFAULT_CDN_BASE_URL


class UploadExample(ExampleBase):
    name = "upload"
    description = "上传文件示例"
    usage = "upload --file <文件路径> --to <用户ID> --token <令牌>"

    def execute(self, args: list) -> int:
        self.print_header("上传文件示例")

        file_path = self.parse_argument(args, "--file")
        to_user_id = self.parse_argument(args, "--to")
        token = self.parse_argument(args, "--token")

        if not file_path or not to_user_id:
            print(f"用法: {self.usage}")
            return 1

        if not os.path.exists(file_path):
            self.print_error(f"文件不存在: {file_path}")
            return 1

        account_store = AccountStore()
        accounts = account_store.list_account_ids()
        if not token and accounts:
            account = account_store.load_account(accounts[0])
            if account: token = account.token
        if not token:
            self.print_error("未找到认证令牌")
            return 1

        client = WeixinApiClient(WeixinApiClientOptions(base_url=DEFAULT_BASE_URL, token=token))
        upload_service = MediaUploadService(client, DEFAULT_CDN_BASE_URL)

        try:
            print(f"正在上传文件: {file_path}")
            uploaded = upload_service.upload_image(file_path, to_user_id)
            self.print_success("上传成功！")
            print(f"  文件键: {uploaded.file_key}")
            print(f"  下载参数: {uploaded.download_encrypted_query_param[:40]}...")
            print(f"  文件大小: {uploaded.file_size} 字节")
            print(f"  密文大小: {uploaded.file_size_ciphertext} 字节")
            return 0
        except Exception as ex:
            self.print_error(f"上传失败: {ex}")
            return 1

File: CSharp/ILinkai.Weixin.Sample/Examples/UploadExample.cs
using Microsoft.Extensions.Logging;
using ILinkai.Weixin.Sdk;
using ILinkai.Weixin.Sdk.Auth;
using ILinkai.Weixin.Sdk.Media;

namespace ILinkai.Weixin.Sample.Examples;

public class UploadExample : ExampleBase
{
    public override string Name => "upload";
    public override string Description => "上传文件示例";
    public override string Usage => "upload --file <文件路径> --to <用户ID> --token <令牌>";

    public UploadExample(ILoggerFactory loggerFactory) : base(loggerFactory) { }

    public override async Task<int> ExecuteAsync(string[] args)
    {
        PrintHeader("上传文件示例");

        var filePath = ParseArgument(args, "--file");
        var toUserId = ParseArgument(args, "--to");
        var token = ParseArgument(args, "--token");

        if (string.IsNullOrEmpty(filePath) || string.IsNullOrEmpty(toUserId))
        {
            Console.WriteLine($"用法: {Usage}");
            return 1;
        }

        if (!File.Exists(filePath))
        {
            PrintError($"文件不存在: {filePath}");
            return 1;
        }

        var accountStore = new AccountStore();
        var accounts = accountStore.ListAccountIds();

        if (string.IsNullOrEmpty(token) && accounts.Count > 0)
        {
            var account = accountStore.LoadAccount(accounts[0]);
            token = account?.Token;
        }

        if (string.IsNullOrEmpty(token))
        {
            PrintError("未找到认证令牌");
            return 1;
        }

        var apiClient = new WeixinApiClient(new WeixinApiClientOptions
        {
            BaseUrl = WeixinApiClient.DefaultBaseUrl,
            Token = token
        }, LoggerFactory.CreateLogger<WeixinApiClient>());

        var uploadService = new MediaUploadService(
            apiClient,
            WeixinApiClient.DefaultCdnBaseUrl,
            LoggerFactory.CreateLogger<MediaUploadService>());

        try
        {
            Console.WriteLine($"正在上传文件: {filePath}");
            var uploaded = await uploadService.UploadImageAsync(filePath, toUserId);

            PrintSuccess("上传成功！");
            Console.WriteLine($"  文件键: {uploaded.FileKey}");
            Console.WriteLine($"  下载参数: {uploaded.DownloadEncryptedQueryParam.Substring(0, Math.Min(40, uploaded.DownloadEncryptedQueryParam.Length))}...");
            Console.WriteLine($"  文件大小: {uploaded.FileSize} 字节");
            Console.WriteLine($"  密文大小: {uploaded.FileSizeCiphertext} 字节");

            return 0;
        }
        catch (Exception ex)
        {
            PrintError($"上传失败: {ex.Message}");
            return 1;
        }
    }
}
